amazon/ion/ioncmodule.c

```diff
@@ -1037,6 +1009,34 @@ static iERR ionc_read_into_container(hREADER hreader, PyObject* container, BOOL
     iRETURN;
 }
 
+/*
+ *  Adds an element to a List or struct
+ *
+ *  Args:
+ *      pyContainer:  A container that the element is added to
+ *      element:  The element to be added to the container
+ *      in_struct:  if the current state is in a struct
+ *      field_name:  The field name of the element if it is inside a struct
+ */
+static void ionc_add_to_container(PyObject* pyContainer, PyObject* element, BOOL in_struct, ION_STRING* field_name) {
+    if (in_struct) {
+        // this builds the "hash-map of lists" structure that the IonPyDict object
+        // expects for its __store
+        PyObject* py_field_name = ion_build_py_string(field_name);
+        PyObject* empty = PyList_New(0);
+        // SetDefault performs get|set with a single hash of the key
+        PyObject* found = PyDict_SetDefault(pyContainer, py_field_name, empty);
+        PyList_Append(found, element);
+
+        Py_XDECREF(py_field_name);
+        Py_XDECREF(empty);
```

Comment:
Was this a conscious decision to change from `Py_XDECREF` from `Py_DECREF`?

Reply:
Semi-conscious? I just noted that it is safe to call pointers that may be null and saw it was used already below. I hadn't fully considered the why or why not.

Reply:
Technically those methods may return NULL pointers, so one would need to use `Py_XDECREF` _if_ we're not checking for NULL returns. Of course if they do return NULL our process is probably cooked anyway.

I'm not super keen to add error checking everywhere so I think using `Py_DECREF` is likely the best choice.